File: todo-app/todo-app-flutter/lib/graphQLdata.dart
String addTask() {
  return ("""mutation AddTask(\$task: [AddTaskInput!]!) {
      addTask(input: \$task) {
        task {
          id
          title
        }
      }
    }""");
}

String fetchAll() {
  return ("""query QueryTask{
      queryTask {
        id
        title
        completed
      }
     }""");
}

String toggleTask() {
  return """mutation ToggleTask(\$taskID: ID!, \$completed: Boolean!) {
      updateTask(input: {
        filter: { id: [\$taskID] },
        set: {
            completed: \$completed
        }
      }) {
        task {
          id
          title
          completed
        }
      }
    }""";
}

String toggleAllTask() {
  return """mutation ToggleAllTask(\$completed: Boolean!) {
      updateTask(input: {
        filter: {},
        set: {
          completed: \$completed
        }
      }) {
        task {
          id
          title
          completed
        }
      }
    }""";
}

String deleteTask() {
  return """mutation DeleteTask(\$taskID: [ID!]) {
      deleteTask(filter: { id: \$taskID }) {
        msg
      }
    }""";
}

String updateTask() {
  return """mutation UpdateTask(\$taskID: ID!, \$task: TaskPatch!) {
      updateTask(input: {
        filter: { id: [\$taskID] },
        set: \$task
      }) {
        task {
          id
          title
          completed
        }
      }
    }""";
}

String clearCompletedTask() {
  return """mutation ClearCompletedTask(\$completed: Boolean) {
      deleteTask(filter: { completed: \$completed }) {
        msg
      }
    }""";
}

File: todo-app/todo-app-vue/src/GraphQLData.js
import gql from "graphql-tag";

export const TASK_MUTATION = gql`
    mutation addTask($task: [AddTaskInput!]!) {
        addTask(input: $task) {
        task {
            id
            title
            completed
        }
        }
    }
`;

export const TASK_QUERY = gql`
    query {
        queryTask {
        id
        title
        completed
        }
    }
`;

export const TASK_DELETE = gql`
    mutation deleteTask($taskID: [ID!]) {
        deleteTask(filter: { id: $taskID }) {
        msg
        }
    }
`;

export const TASK_TOGGLE = gql`
    mutation updateTask($taskID: ID!, $completed: Boolean!) {
        updateTask(input: {
            filter: { id: [$taskID] },
            set: {
                completed: $completed
            }
        }) {
            task {
                id
                title
                completed
            }
        }
    }
`;

export const TASK_EDIT = gql`
    mutation updateTask($id: ID!, $title: String! ) {
        updateTask(input: {
            filter: { id: [$id] },
            set: {
                title: $title
            }
        }) {
            task {
                id
                title
                completed
            }
            numUids
        }
    }
`;

export const CLEAR_COMPLETED_TASKS = gql`
    mutation clearCompletedTask($completed: Boolean) {
        deleteTask(filter: { completed: $completed }) {
            msg
        }
    }
`;

export const TOGGLE_ALL_TASKS = gql`
    mutation toggleAllTasks($completed: Boolean!) {
        updateTask(input: {
            filter: {},
            set: {
                completed: $completed
            }
        }) {
            task {
                id
                title
                completed
            }
        }
    }
`;